### PR TITLE
update gradle 6.0.1 & fix buildAll, uploadArchivesAll on gradle 6.x. Fix #46

### LINKED ISSUE #46: ./gradlew clean build_2.12.10 failed with gradle 6.x
## Expected Behavior
test pass
## Actual Behavior
test failed. See the following comment below

## Steps to Reproduce
```bash
./gradlew clean build_2.12.10
```
## Possible Solution
update mokito 3.2.0

## Your Environment
* Reactor version(s) used: commit 9c2b022
* gradle version: 6.0.1 (see in file gradle/wrapper/gradle-wrapper.properties in the [tree](https://github.com/reactor/reactor-scala-extensions/tree/9c2b022) of commit 9c2b022)
* Other relevant libraries versions (eg. `netty`, ...): none
* JVM version (`java -version`):
openjdk version "13.0.1" 2019-10-15
OpenJDK Runtime Environment AdoptOpenJDK (build 13.0.1+9)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 13.0.1+9, mixed mode, sharing)
* OS and version (eg `uname -a`):
19.0.0 Darwin Kernel Version 19.0.0: Thu Oct 17 16:17:15 PDT 2019; root:xnu-6153.41.3~29/RELEASE_X86_64 x86_64

### PR DESCRIPTION
**Pull request by giabao**
also update travis to run buildAll, uploadArchivesAll

### CODE REVIEW

**Comment by giabao** [comment]
ok @sinwe 
I have rolled back to multi jobs build and add a job to run `buildAll` to verify that task work on gradle 6

**Comment by sinwe** [comment]
@giabao 
Or you just pull my master to your branch and upgrade gradle and apply the buildName fix

File: .travis.yml [hunk 1]
```diff
@@ -12,34 +12,30 @@ env:
...
     - secure: MlwQtz4tO1mmNmPE2PIxYI552puyQpQa690C6ZOldvL0zGhw5mgj5JD8G4I4RmuaNev7mpdnBecoTEg6q2pxPTHQEgQmHVX5X7SPf1Loyes35AWzscGhWVwFdOF6ZqccB8PUZrUZ2CAD83JjG39/PGk5bxqwBNFkeFBQrGYhDuL5caqJ9/MTfKHM/NoR66ygigscnsfb8qwtS1wpSJ5LbFrC2KFzGfDXuP5bjkz9PZdQrS8c7nC63KcdeENpqwdrWWzrw3hf0uSh3xEZp7SziDZ0wMOX6lmIIzRjmlJIPF4BD9MsDiPqhmwJk4MdvKtWAzKJTnGxmygNLVw7YKnqULnWjsOmNfraHGEkqSE/DCccsjAIzYP0vD7V+JgXOk5JLvwLGff3VW9457f2pAMWJgliL9r136DKLMvxiMpXgFp3BkZHsc80Ve1G9nqgGMqp1zrCgE78mqQJ5AFQczfTZOXNsjLkGYgs38k7aLiCAm8p9ZHYBo4DZiaPKc/K2XkBmTP+2KuHpwXGUbnvfFFoHuwV1iDh+CeEo1N4UZrhwPCXTNaJKYQCQ8d7u2Vvt27Ido+IfvSnKvBlHMkpmzAdyF5dhT3jR0xEartpMw8MmUnKVxanJitfLME1qInf7INhFkkRECj9yFpFA1kCM+dszxdEEaBmD0D004xd0rU4Y3c=
-script:
-  - ./gradlew build --stacktrace
+jobs:
```
Comment (sinwe) [inline .travis.yml]: I think this don't need to be changed. Just leave it as it was before

File: .travis.yml [hunk 2]
```diff
@@ -12,34 +12,30 @@ env:
...
+  include:
+  - name: "buildAll checkScoverage"
+    jdk: openjdk11
   # scoverage 4.0.1 need scala 2.12
   #  Execution failed for task ':reportTestScoverage'.
   #> 'scala.collection.convert.Decorators$AsScala scala.collection.JavaConverters.asScalaBufferConverter(java.util.List)'
   #Caused by: java.lang.NoSuchMethodError: 'scala.collection.convert.Decorators$AsScala scala.collection.JavaConverters.asScalaBufferConverter(java.util.List)'
   #
   # see https://github.com/scoverage/gradle-scoverage/pull/130
-  - |
```
Comment (sinwe) [inline .travis.yml]: This too, no need to be removed

File: .travis.yml [hunk 3]
```diff
@@ -12,34 +12,30 @@ env:
...
-after_success:
-  - |
-    if [[ "$TRAVIS_JDK_VERSION" == "openjdk11" && "$ORG_GRADLE_PROJECT_scalaVersion" == "2.12.10" ]]; then
-      bash <(curl -s https://codecov.io/bash) || echo "Codecov did not collect coverage reports"
-    fi
-
-deploy:
-  provider: script
-  cleanup: false
-  script: ./gradlew uploadArchives --stacktrace
```
Comment (sinwe) [inline .travis.yml]: Here just do `./gradlew uploadArchiveAll --stacktrace` like it was before

File: build.gradle [hunk 4]
```diff
@@ -54,19 +54,25 @@ if (!project.parent) {
...
     final svs = ['2.12.10', '2.11.12', '2.13.1']
 
-    for (sv in svs) {
-        String scalaVersionInDot = sv.replaceAll("_", ".")
-
+    svs.eachWithIndex { sv, i ->
         tasks.create(name: "build_$sv", type: GradleBuild) {
+// set a different buildName for each build task so gradle 6's DefaultIncludedBuildRegistry.assignPath don't throw
+// GradleException("Included build " + dir + " has build path " + requestedPath + " which is the same as included build " + existingForPath)
+            buildName = "build_${sv}"
```
Comment (sinwe) [inline build.gradle]: Nice 👍

**Comment by sinwe** [comment]
I've selected changes from your PR and applied it to master.
So I'll close this PR. Thanks for your PR and contribution 👍

**Pull request closed without merging**